dagger/main.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A minimal example of how to include Prometheus instrumentation.
+
+package main
+
+import (
+	"context"
+	"strings"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type ClientGolang struct {
+	Source *Directory // +private
+}
+
+func New(src *Directory) *ClientGolang {
+	return &ClientGolang{Source: src}
+}
+
+// runs `make` with the given arguments
+func (m *ClientGolang) Make(
+	// +optional
+	args string,
+	// +default="1.20"
+	goVersion string,
+	// +optional
+	env []string,
+) (string, error) {
+	return dag.Golang().
+		Base(goVersion).
+		Container().
+		WithMountedDirectory("/src", m.Source).
+		WithWorkdir("/src").
+		WithMountedCache("/go/bin", dag.CacheVolume("gobincache")).
+		WithExec([]string{"sh", "-c", "make " + args}).
+		Stdout(context.Background())
+}
+
+// runs `make` with the given arguments for all supported go versions
+func (m *ClientGolang) MakeAll(
```

Comment:
I think `make all` is a bit excessive to run in this repo. We can live with
- `make test`
- `make fmt`
for starters.

This could also solve the issue with the license checker that you had to edit above.

Reply:
And maybe `make lint`

Reply:
Let's call it `MakeLintTest` then?

Reply:
Ok, yes, the name is confusing, then. `MakeRun` or `MakeRunWithSupportedVersions` would make more sense to me.